runtime/compiler/z/codegen/UncommonBCDCHKAddressNode.cpp

```diff
@@ -50,37 +50,61 @@ UncommonBCDCHKAddressNode::perform()
       if(node && node->getOpCodeValue() == TR::BCDCHK && node->getSymbol() && node->getSymbol()->getResolvedMethodSymbol())
          {
          TR::RecognizedMethod method = node->getSymbol()->getResolvedMethodSymbol()->getRecognizedMethod();
-         if(method == TR::com_ibm_dataaccess_DecimalData_convertIntegerToPackedDecimal_ ||
-            method == TR::com_ibm_dataaccess_DecimalData_convertLongToPackedDecimal_ ||
-            method == TR::com_ibm_dataaccess_PackedDecimal_addPackedDecimal_ ||
-            method == TR::com_ibm_dataaccess_PackedDecimal_subtractPackedDecimal_ ||
-            method == TR::com_ibm_dataaccess_PackedDecimal_multiplyPackedDecimal_ ||
-            method == TR::com_ibm_dataaccess_PackedDecimal_dividePackedDecimal_ ||
-            method == TR::com_ibm_dataaccess_PackedDecimal_remainderPackedDecimal_||
-            method == TR::com_ibm_dataaccess_PackedDecimal_shiftLeftPackedDecimal_ ||
-            method == TR::com_ibm_dataaccess_PackedDecimal_shiftRightPackedDecimal_)
+         TR::Node* oldAddressNode = node->getSecondChild();
```

Comment:
TR::BCDCHK has unspecified child properties. Are we guaranteed to have second child?

Reply:
I have updated the if condition to enter only if the node BCDCHK has two children.